tests/TestsExternUniformPowerOfTwo.dfy

```diff
@@ -106,4 +107,28 @@ module TestsExternUniform {
     var r := new DafnyVMC.DRandomExternUniformPowerOfTwo();
     Tests.TestDiscreteGaussian(1_000_000, r);
   }
+
+  method {:test} TestFisherYatesIdentity()
+    decreases *
+  {
+    var a: array<nat> := new nat[4](i => i); // [0, 1, 2, 3]
+    var r := new DafnyVMC.DRandomExternUniformPowerOfTwo();
+    Tests.TestFisherYates(1_000_000, a, r, Helper.NatToString);
+  }
+
+  method {:test} TestFisherYatesConstant()
+    decreases *
+  {
+    var a: array<nat> := new nat[4](i => 0); // [0, 0, 0, 0]
+    var r := new DafnyVMC.DRandomExternUniformPowerOfTwo();
+    Tests.TestFisherYates(1_000_000, a, r, Helper.NatToString);
+  }
+
+  method {:test} TestFisherYatesMixed()
+    decreases *
+  {
+    var a: array<nat> := new nat[] [0, 1, 1, 2]; // [0, 1, 1, 2]
+    var r := new DafnyVMC.DRandomExternUniformPowerOfTwo();
+    Tests.TestFisherYates(1_000_000, a, r, Helper.NatToString);
+  }
 }
```

Comment:
Same as below. We could also do some differential testing against the Java implementation.